app/services/arcgis_api/geocoder.rb

```diff
@@ -59,8 +59,14 @@ def find_address_candidates(magic_key)
     # it expires after 1 hour
     # @return [String] Auth token
     def retrieve_token!
-      token, expires = request_token.fetch_values('token', 'expires')
-      Rails.cache.write(API_TOKEN_CACHE_KEY, token, expires_at: Time.zone.at(expires))
+      token, expires_at = request_token.fetch_values('token', 'expires')
+      expires_in = Time.zone.at(expires_at) - Time.zone.now
+      Rails.cache.write(API_TOKEN_CACHE_KEY, token, expires_in: expires_in)
+      # If using a redis cache we have to manually set the expires_in. This is because we aren't
+      # using a dedicated Redis cache and instead are just using our existing Redis server with
+      # mixed usage patterns. Without this cache entries don't expire.
+      # More at https://api.rubyonrails.org/classes/ActiveSupport/Cache/RedisCacheStore.html
```

Comment:
Wow, that is extremely unintuitive. What is “expiring” then, it will just write over it without reading?

Reply:
I think the class assumes a fixed redis server-side eviction policy

Without it, the entries will live forever?

Reply:
Correct, entries seem to just live forever